syft/generic/frameworks/hook/hook.py

```diff
@@ -271,6 +273,73 @@ def _hook_multi_pointer_tensor_methods(self, tensor_type):
                 new_method = self._get_hooked_multi_pointer_method(attr)
                 setattr(MultiPointerTensor, attr, new_method)
 
+    def _hook_string_methods(self, owner):
+
+        # Set the default owner
+        setattr(String, "owner", owner)
+
+        # Set of methods to hook/override for 'str' to String
+        attr_include = set(
+            [
+                "__add__",
+                "__eq__",
+                "__le__",
+                "__ge__",
+                "__gt__",
+                "__lt__",
+                "__ne__",
+                "__len__",
+                "__getitem__",
+                "__str__",
+                "__repr__",
+                "__format__",
+                "lower",
+                "upper",
```

Comment:
Here is a short list of all str method in python, you might want to include more like title or capitalize maybe depending on your usecases: https://www.w3schools.com/python/python_ref_string.asp

Reply:
Done